### PR TITLE
added tvOS support

### PR DESCRIPTION
**Pull request by yarneo**
tvOS support + tvOS demo project + tvOS demo project tests

### CODE REVIEW

File: Mixpanel/Mixpanel.m [hunk 1]
```diff
@@ -150,7 +154,7 @@ - (instancetype)initWithToken:(NSString *)apiToken launchOptions:(NSDictionary *
         MixpanelDebug(@"%@ warning empty api token", self);
     }
     if (self = [self init]) {
-#if !defined(MIXPANEL_APP_EXTENSION)
+#if !defined(MIXPANEL_APP_EXTENSION) && !defined(MIXPANEL_TVOS_EXTENSION)
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Might be good to keep this one. Does tvOS support exception handlers?

**Comment by samgreen** [comment]
Some minor comments throughout. Since we're now support four different integrations (iOS, watchOS, tvOS, and App Extensions) I think it might be useful to start breaking parts of this file out. As it is now, adding support for tvOS results in half of the file being removed by the preprocessor, it's a cognitive load that makes things a bit harder to work with.

I'm also curious if we can get the tvOS tests to share the same pods as the iOS tests. No need to duplicate those.

**Comment by samgreen** [comment]
It looks like you may have accidentally committed the pods.

**Comment by yarneo** [comment]
A new PR will be open from a new branch

**Pull request closed without merging**